python/bin/watchman-wait-aio

```diff
@@ -0,0 +1,222 @@
+is a single event.  You may also remove the limit and allow it to execute
+continuously.
+
+watchman-wait will print one event per line.  The event information includes
+your specified list of fields, with each field separated by a space (or your
+choice of --separator).
+
+Events are consolidated and settled by the watchman server before they are
+dispatched to watchman-wait.
+
+Exit Status:
+
+The following exit status codes can be used to determine what caused
+watchman-wait to exit:
+
+0  After successfully waiting for event(s)
+1  In case of a runtime error of some kind
+2  The -t/--timeout option was used and that amount of time passed
+   before an event was received
+3  Execution was interrupted (Ctrl-C)
+
+''')
+parser.add_argument('path', type=str, nargs='+',
+                    help='path(s) to watch')
+parser.add_argument('--relative', type=str, default='.',
+                    help='print paths relative to this dir (default=PWD)')
+parser.add_argument('--fields', type=fieldlist, default=['name'],
+                    help='''
+Comma separated list of file information fields to return.
+The default is just the name.  For a list of possible fields, see:
+https://facebook.github.io/watchman/docs/cmd/query.html#available-fields
+''')
+parser.add_argument('-s', '--separator', type=str, default=' ',
+                    help='String to use as field separator for event output.')
+parser.add_argument('-m', '--max-events', type=int, default=1,
+                    help='''
+Set the maximum number of events that will be processed.  When the limit
+is reached, watchman-wait will exit.  The default is 1.  Setting the
+limit to 0 removes the limit, causing watchman-wait to execute indefinitely.
+''')
+parser.add_argument('-p', '--pattern', type=str, nargs='+',
+                    help='''
+Only emit paths that match this list of patterns.  Patterns are
+applied by the watchman server and are matched against the root-relative
+paths.
+
+You will almost certainly want to use quotes around your pattern list
+so that your shell doesn't interpret the pattern.
+
+The pattern syntax is wildmatch style; globbing with recursive matching
+via '**'.
+''')
+parser.add_argument('-t', '--timeout', type=float, default=0, help='''
+Exit if no events trigger within the specified timeout.  If timeout is
+zero (the default) then keep running indefinitely.
+''')
+args = parser.parse_args()
+
+
+# We parse the list of paths into a set of subscriptions
+subscriptions = {}
+
+# Running total of individual file events we've seen
+total_events = 0
+
+
+class Subscription(object):
+    root = None     # Watched root
+    relpath = None  # Offset to dir of interest
+    name = None     # Our name for this subscription
+    path = None
+
+    def __init__(self, name, path):
+        self.name = name
+        self.path = path
+
+    def __repr__(self):
+        return 'Subscription(root=%s, rel=%s, name=%s)' % (
```

Comment:
I changed the `__repr__` to `__str__`, but also made `__init__` take relpath. There also are other changes due to restructuring, so please take a look.